libs/vkd3d-shader/spirv.c

```diff
@@ -6964,7 +6964,10 @@ static void vkd3d_dxbc_compiler_emit_imad(struct vkd3d_dxbc_compiler *compiler,
     unsigned int i, component_count;
 
     component_count = vkd3d_write_mask_component_count(dst->write_mask);
-    type_id = vkd3d_spirv_get_type_id(builder, VKD3D_TYPE_INT, component_count);
+    if (instruction->handler_idx == VKD3DSIH_UMAD)
+        type_id = vkd3d_spirv_get_type_id(builder, VKD3D_TYPE_UINT, component_count);
+    else
+        type_id = vkd3d_spirv_get_type_id(builder, VKD3D_TYPE_INT, component_count);
```

Comment:
Can we use `vkd3d_dxbc_compiler_get_type_id_for_dst` here?

Reply:
Yes. Done.